src/mainboard/pcengines/apu2/mainboard.c

```diff
@@ -433,6 +433,61 @@ const char *smbios_mainboard_sku(void)
 	return sku;
 }
 
+static int fill_mainboard_smbios_type16(unsigned long *current, int *handle)
+{
+	// read SPD info from file
+	spd_raw_data *spd;
+	void *spd_file;
+	size_t spd_file_len = 0;
+	spd_file = cbfs_boot_map_with_leak("spd.bin", CBFS_TYPE_SPD, &spd_file_len);
```

Comment:
@finus16 look: https://github.com/pcengines/coreboot/blob/ec6dcc3ecbb009984a42b9d5346aa84f20e43212/src/mainboard/pcengines/apu2/mainboard.c#L221
This function will get You the correct 128 byte SPD from CBFS according to mainboard configuration. Why bothering with memcpy?

Reply:
corrected